File: ABC238/B.py
# 入力
N = int(input())
A = list(map(int, input().split()))

# ピザのカット位置
B = [0]*(N+1)
for i in range(N):
  B[i+1] = B[i] + A[i]
  # B[i+1]が360を超えている場合は、一周して360以内に戻す
  if B[i+1] > 360:
    B[i+1] -= 360

B = sorted(B)
# 計算の都合上、最後に360を追加
B.append(360)

# 最大のピザを取得
ans = 0
for i in range(N+1):
  ans = max(ans, B[i+1]-B[i])

print(ans)

File: ABC238/B.cpp
#include <bits/stdc++.h>
#define rep(i, m, n) for(int i = m; i < n; i++)
#define INF 2e9
#define all(v) v.begin(), v.end()
using namespace std;
typedef long long ll;

int main()
{
  int N;
  cin >> N;
  vector<int> A(N);
  rep (i, 0, N) cin >> A.at(i);

  vector<int> B(N+1);
  rep (i, 0, N) {
    B.at(i+1) = B.at(i) + A.at(i);
    if (B.at(i+1) > 360) B.at(i+1) -= 360;
  }
  sort(all(B));
  B.push_back(360);
  // rep (i, 0, N+2) printf("%d %d\n", i, B.at(i));
  // rep (i, 1, N+2) cout << B.at(i) - B.at(i-1) << endl;
  vector<int> C(N+1);
  int ans = 0;
  rep (i, 1, N+2) {
    ans = max(ans, B.at(i) - B.at(i-1));
  }
  cout << ans << endl;

}
